signer/signer.go

```diff
@@ -36,24 +36,24 @@ import (
 // signingAgent is the unprotected header field used by signature.
 const signingAgent = "Notation/1.0.0"
 
-// genericSigner implements notation.Signer and embeds signature.Signer
-type genericSigner struct {
+// GenericSigner implements notation.Signer and embeds signature.Signer
+type GenericSigner struct {
 	signature.Signer
 }
 
 // New returns a builtinSigner given key and cert chain
-func New(key crypto.PrivateKey, certChain []*x509.Certificate) (notation.Signer, error) {
+func New(key crypto.PrivateKey, certChain []*x509.Certificate) (*GenericSigner, error) {
```

Comment:
Changing the function signature is a breaking change. Please consider a new constructor method.

Reply:
we are moving from interface to a particular implementation. Shouldn't this be backward compatible ?